client/src/components/PredictionPanel.vue

```diff
@@ -1,39 +1,88 @@
 <template>
   <div class='container'>
-    <map-geocoder-wrapper v-if='showSearch'
-                          v-model:expandSearch='expandSearch' />
-    <div class='lead-prediction' v-if='showPrediction'>
-      You have a high likelihood of lead.
+    <div class='explain-text'>
+      {{ ScorecardSummaryMessages.LEADOUT }}
+    </div>
+    <map-geocoder-wrapper v-model:expandSearch='showSearch' />
+    <div class='h1-header semi-bold'
+         v-if='this.pwsId != null && percentLead != null'>
+      Your neighborhood water system {{ pwsId }} is likely
+      {{ percentLead }}% lead.
+    </div>
+    <div class='explain-text'>
+      {{ ScorecardSummaryMessages.LEAD_LIKELIHOOD_EXPLAINED }}
     </div>
   </div>
 </template>
 
 <script lang='ts'>
 import { defineComponent } from 'vue';
 import MapGeocoderWrapper from './MapGeocoderWrapper.vue';
+import { dispatch, useSelector } from '../model/store';
+import { ScorecardSummaryMessages } from '../assets/messages/scorecard_summary_messages';
+import { getWaterSystem } from '../model/slices/lead_data_slice';
+import { GeoState } from '../model/states/geo_state';
+import { LeadDataState } from '../model/states/lead_data_state';
 
 /**
  * Container lead prediction.
  */
 export default defineComponent({
   name: 'PredictionPanel',
   components: { MapGeocoderWrapper },
+  setup() {
+    // Listen to state updates.
+    const geoState = useSelector((state) => state.geos) as GeoState;
+    const leadDataState = useSelector((state) => state.leadData) as LeadDataState;
+
+    return {
+      geoState: geoState,
+      leadState: leadDataState,
```

Comment:
out of curiosity - so generally this is how we want to listen to store changes? and no components will look at store directly, this is hidden in useSelector?

Reply:
This seems to be a suggested way of doing it to avoid duplicated code but this "as XState" type casting is not ideal. The issue is we don't have the react hooks that give these states back typed